services/cadip/rs_server_cadip/cadip_retriever.py

```diff
@@ -0,0 +1,79 @@
+"""This will be used to configure dataretriever used for cadip"""
+import json
+import os.path as osp
+from pathlib import Path
+
+from services.common.rs_server_common.data_retrieval.data_retriever import DataRetriever
+from services.common.rs_server_common.data_retrieval.download_monitor import (
+    DownloadMonitor,
+)
+from services.common.rs_server_common.data_retrieval.provider import (
+    CreateProviderFailed,
+    Provider
+)
+from services.common.rs_server_common.data_retrieval.storage import Storage
+
+CONF_FOLDER = Path(osp.realpath(osp.dirname(__file__))).parent / "config"
+EODAG_CONFIG = Path(osp.realpath(osp.dirname(__file__))).parent / "config" / "cadip_ws_config.yaml"
+
+
+def station_to_server_url(station: str) -> str | None:
+    """Retrieve the configuration data (webserver address) for a CADU station based on its identifier.
+
+    Parameters
+    ----------
+    station : str
+        Identifier for the CADU station.
+
+    Returns
+    -------
+    str or None
+        A str containing the webserver address for the specified station,
+        or None if the station identifier is not found.
+
+    Example
+    -------
+    >>> station_to_server_url("station123")
+    'https://station123.example.com'
+
+    Notes
+    -----
+    - The station identifier is case-insensitive and is converted to uppercase for matching.
+    - The function reads the station configuration data from a JSON file.
+    - If the station identifier is not found in the configuration data, the function returns None.
+    """
+    try:
+        with open(CONF_FOLDER / "stations_cfg.json", encoding="utf-8") as jfile:
+            stations_data = json.load(jfile)
+            return stations_data.get(station.upper(), None)
+    except (FileNotFoundError, json.JSONDecodeError):
+        # logger to be added.
+        return None
+
+
+def init_cadip_data_retriever(cadip_provider: Provider, station: str):
```

Comment:
The cadip_provider isn't useful.
To initialize the data retriever for CADIP,
we want to use an EodagProvider.
You don't need to parameterize it.

Reply:
Agree.